File: GestionEvenementBack/src/main/java/tn/eesprit/gestionevenementback/Entities/Post.java
package tn.eesprit.gestionevenementback.Entities;

import com.fasterxml.jackson.annotation.JsonIgnore;
import javax.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.io.Serializable;
import java.util.Date;
import java.util.Set;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
@Entity
public class Post implements Serializable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer postId;
    String content;
    String title;
    Date datePub;
    @ManyToMany
    Set<User> likes;
    @ManyToMany
    Set<User> views;


    @JsonIgnore
    @ManyToOne
    Forum forum;

    @OneToMany(mappedBy = "post")
    Set<Message> messages;

    @ManyToOne
    User user;
}

File: GestionEvenementFront/src/app/_Models/Post.ts
import {Forum} from "./Forum";
import {Message} from "./Message";
import {User} from "./User";

export class Post {
  postId: number;
  title:string;
  content:string;
  forum:Forum ;
  datePub:Date;
  messages:Message[];
  user:User;
  views:User[];
  likes:User[];
}
